zenodo/modules/deposit/api.py

```diff
@@ -245,11 +245,12 @@ def _get_auto_added(self, record):
     def _process_files(self, record_id, data):
         """Snapshot bucket and add files in record during first publishing."""
         if self.files:
-            file_uuids = []
+            file_uuids = set()
             for f in self.files:
                 fs, path = f.file.storage()._get_fs()
-                if not fs.exists(path):
-                    file_uuids.append(str(f.file.id))
+                if not (fs.exists(path) and
+                        f.file.verify_checksum(throws=False)):
```

Comment:
Shouldn't this condition be
```python
if not fs.exists(path) or not f.file.verify_checksum(throws=False):
```
but also assuming also that ``verify_checksum`` fails when file doesn't exist (not sure about that, needs checking), perhaps it should be simply:
```python
if not f.file.verify_checksum(throws=False):
```